### DISCUSSION TITLE
Perimeter 81

### DISCUSSION THREAD

**Discussion by BlythMeister** [chocolatey-community/chocolatey-packages#2363] [Ideas]
We would like to introduce a package for perimeter 81.

We currently manage this for our organisation in a private feed.  But this needs local network access to download.

Since perimeter 81 gives the remote access to local network, you can't install the choco package until you have the software installed.  Which makes it a bit pointless.

We also wanted to leverage the AU package mechanism to keep the version updated.

I have raised a PR for this already (#2362) as I didn't know it needed discussion 1st.

So, here is a retrospective discussion 🙂

**Comment by pauby** [comment]
Is there a reason you cannot maintain this package yourself and submit it to the Chocolatey Community Repository?

**Reply by BlythMeister** [reply]
The reason i submitted the PR as i did was because the whole automated updates thing that is setup here seems to work really well and i thought it was a place for community packages to go being that it's the "chocolatey-community" organisation.

Maybe i've misunderstood how this all works.

**Reply by BlythMeister** [reply]
Are you suggesting i go about setting up all the automation that already exists in the "chocolatey-community" organistation within my own github to push to chocolatey?

Makes the community not seem like much of a community doesn't it?

**Reply by pauby** [reply]
> The reason i submitted the PR as i did was because the whole automated updates thing that is setup here seems to work really well and i thought it was a place for community packages to go being that it's the "chocolatey-community" organisation.

> Maybe i've misunderstood how this all works.

The Chocolatey Community Chocolatey packages repository is a place where a group of community maintainers maintain around 350 packages. These packages have been built-up historically, and they contain many of the popular packages from the Chocolatey Community Repository. The small group of maintainers do this work in their spare time. 

The repository is not a place where you just add a package and leave the community to maintain it. You're expected to maintain the package. Even though the package is automated, there will still be maintenance required when things break.

I'm not suggesting you were intending to do that, but it sounds like you haven't read the submission guidelines, know what the community is about and didn't add your name to the CODEOWNERS file to say you would maintain the package going forward.

**Reply by pauby** [reply]
> Are you suggesting i go about setting up all the automation that already exists in the "chocolatey-community" organistation within my own github to push to chocolatey?

As this package is primarily for your organisation, that would make most sense if the package isn't accepted here. That's what most other maintainers have done.

> Makes the community not seem like much of a community doesn't it?

It would probably feel better if you understood what the Community is about, what this repository is for and read the guidelines and documentation we have in here.

In addition, that comment isn't helpful.

**Reply by BlythMeister** [reply]
for the record I read this https://github.com/chocolatey-community/chocolatey-packages/blob/master/CONTRIBUTING.md

Didn't see anything on there about not creating new packages, having to raise discussions before doing so or having to add yourself to the CODEOWNERS.

Perhaps if this was clear i would of done so rather than getting a feeling that i'm being told off by a maintainer...

It's not necessarily something only used by my organisation, Perimeter81 is a product that anyone can use.

**Reply by AdmiringWorm** [reply]
A code owner for the package is required, this is mentioned under the `Maintainers` header: https://github.com/chocolatey-community/chocolatey-packages/blob/master/CONTRIBUTING.md#maintainers

For the issue one: this statement is mentioned under the `Opening Pull Requests` header.
> Pull requests fixing a bug or an outdated package do not need to have an associated issue.

This means anything that isn't related to a bug, or an outdated package **MUST** have an issue associated. We do not mention anything specific about adding Pull Requests for new package, only for migrating packages as that is what we generally allow in this repository.

**Comment by pauby** [comment]
Perhaps @AdmiringWorm would comment on this package?

**Comment by BlythMeister** [comment]
Given up on the idea.

**Reply by BlythMeister** [reply]
to Elaborate, i'm going to host myself as the community packages seems to be a iron fenced non-community managed package set.

**Reply by pauby** [reply]
I want to suggest you look back at your actions and comments in here and reflect on how you are coming across. Given your position, I find unhelpful and unusable comments such as these unexpected and disappointing.

I'm going to go and ahead and lock this discussion, as I don't believe anything useful can come out of it. If you wish to be involved in the community in the future, please ensure you follow the [Code Of Conduct](https://github.com/chocolatey-community/.github/blob/main/CODE_OF_CONDUCT.md).

**Comment by AdmiringWorm** [comment]
We generally do not accept entirely new packages into this repository (there are some exceptions, but these happen on a case by case basis).

However, we sometimes do accept migrations of existing packages.

To have the community maintain a package for your software, you should create an issue over at: https://github.com/chocolatey-community/chocolatey-package-requests/issues

After someone picks up the issue and has pushed their first package version (and it is approved), a request to migrate the package to this repository can be made through the discussion template: https://github.com/orgs/chocolatey-community/discussions/new?category=migrations

Once that discussion is approved for migration, it can be migrated to this repository.

Do take note of the priority we have listed in our [contribution document](https://github.com/chocolatey-community/chocolatey-packages/blob/master/CONTRIBUTING.md), as it is unlikely something that falls outside of those priorities will be accepted here.
